micromasters/serializers.py

```diff
@@ -0,0 +1,78 @@
+"""
+Serializers for Django contrib models
+"""
+import logging
+from rest_framework import serializers
+from django.contrib.auth.models import User, AnonymousUser
+from django.core.exceptions import ObjectDoesNotExist
+from rolepermissions.verifications import has_role
+from roles.models import Instructor, Staff
+from backends.edxorg import EdxOrgOAuth2
+
+log = logging.getLogger(__name__)
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """Serializer for users."""
+    first_name = serializers.SerializerMethodField()
+    last_name = serializers.SerializerMethodField()
+    preferred_name = serializers.SerializerMethodField()
+    is_staff = serializers.SerializerMethodField()
+    social_username = serializers.SerializerMethodField()
+
+    class Meta:
+        model = User
+        fields = (
+            "username", "email", "social_username",
+            "first_name", "last_name", "preferred_name",
+            "is_staff",
+        )
+
+    def get_first_name(self, obj):  # pylint: disable=no-self-use
+        "Get first_name from user profile"
+        if obj.profile:
```

Comment:
If this fails a `DoesNotExist` exception is thrown, so the `if` expression will never be false. Can you change this and the ones below to catch the `DoesNotExist` instead?